Content.Server/Chemistry/Components/SolutionRegenerationComponent.cs

```diff
@@ -0,0 +1,45 @@
+using Content.Server.Chemistry.EntitySystems;
+using Content.Shared.Chemistry.Reagent;
+using Content.Shared.FixedPoint;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+
+namespace Content.Server.Chemistry.Components;
+
+/// <summary>
+/// Passively increases a solution's quantity of a reagent.
+/// </summary>
+[RegisterComponent]
+[Access(typeof(SolutionRegenerationSystem))]
+public sealed class SolutionRegenerationComponent : Component
+{
+    /// <summary>
+    /// The solution to have reagent added to.
+    /// </summary>
+    [DataField("solution", required: true), ViewVariables(VVAccess.ReadWrite)]
+    public string Solution = string.Empty;
+
+    /// <summary>
+    /// The reagent to be regenerated in the solution.
+    /// </summary>
+    [DataField("reagent", required: true, customTypeSerializer: typeof(PrototypeIdSerializer<ReagentPrototype>)), ViewVariables(VVAccess.ReadWrite)]
+    public string Reagent = string.Empty;
+
+    /// <summary>
+    /// Quantity of the reagent regenerated every time.
+    /// </summary>
+    [DataField("quantity"), ViewVariables(VVAccess.ReadWrite)]
+    public FixedPoint2 Quantity = FixedPoint2.New(1f);
```

Comment:
Can this be `Solution` instead, so it's not limited to a single reagent?

Reply:
ill see how it gets serialized and try change it yeah